ideas/single_connection_usage_with_timeouts.cpp

```diff
@@ -2,12 +2,13 @@ boost::asio::io_service io_service;
 
 boost::asio::spawn(io_service, [&] (boost::asio::yield_context yield) {
     auto connection = apq::make_connection_provider(io_service);
-    apq::cursor cursor;
+    apq::data_row row;
     connection_timeouts timeouts;
     timeouts.connect = 100ms;
     timeouts.request = 200ms;
-    apq::text_protocol::request(connection, cursor, timeouts, "SELECT pg_is_in_recovery()", yield);
-    std::cout << cursor.next(yield).at(0) << '\n';
+    if (apq::text_protocol::request(connection, apq::make_query("SELECT pg_is_in_recovery()", timeouts), row, yield)) {
```

Comment:
Why timeouts are the part of query?

Reply:
What is the result of request, why we should use it as condition? Besides, there is always one row result in this example.